portfolio/src/main/webapp/script.js

```diff
@@ -95,9 +103,28 @@ function createCommentElement(comment) {
   return commentElement;
 }
 
-/** Tells the server to delete the task. */
+/** Tells the server to delete the comment. */
 function deleteComment(comment) {
   const params = new URLSearchParams();
   params.append('id', comment.id);
   fetch('/delete-data', {method: 'POST', body: params});
 }
+
+/** Tells the server to delete ALL comments */
+async function deleteAllComments(){
+
+  // Get all the comments on the server.
+  const response = await fetch("/data");
+  const comments = await response.json();
+
+  // Delete each comment from the server.
+  var i;
```

Comment:
never use var, always use const or let (let in this situation).
var has a really weird scope, it's a global variable, but scoped to the function so it can only be accessed inside the function but it persists between calls.
also you can declare it inside the for loop eg for(let i = 0;...).
also comments.forEach() is another way of doing this just fyi.

Reply:
Done (updated to a comments.forEach())